# Задание-2.
#
# Описание:
#
# 2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
# Проверьте его работу на данных, вводимых пользователем.
# При вводе пользователем нуля в качестве делителя программа должна корректно
# обработать эту ситуацию и не завершиться с ошибкой.
#
# .


""


class UsrError(Exception):
    def __init__(self,text2usr):
        self.text2usr = text2usr


usr_val_input = input("Введите целое положительное число:__ ")

try:

    if usr_val_input != "0":
        raise UsrError("Ошибка! Вы совершили попытку деления на ноль!")
    else:
        raise UsrError(f"Python знаком с правилами арифметики..., поэтому ноль на ноль не делит ;)")

except UsrError as divzero:
    print(divzero)

